src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -225,8 +221,8 @@ protected virtual Task OnPostActorMethodAsync(ActorMethodContext actorMethodCont
         /// </returns>
         protected Task UnregisterReminderAsync(IActorReminder reminder)
         {
-            EnsureInteractorInitialized();
-            return this.Host.DaprInteractor.UnregisterReminderAsync(this.actorTypeName, this.Id.ToString(), reminder.Name);
+            var token = reminder is ActorReminderToken ? (ActorReminderToken)reminder : new ActorReminderToken(this.actorTypeName, this.Id, reminder.Name);
```

Comment:
This gets a little confusing, hopefully this will clear it up. For both actors and reminders there are a few existing data types in use. None of them really fit the bill for abstracting this functionality. However we're stuck supporting the types that are part the public API.

Ex:

To register a timer you need:

- actor type
- actor id
- name
- callback
- state
- due time
- period

To unregister a timer you need:

- actor type
- actor id
- name

None of the existing types has all of those things for either case 😱 so none of them are useful for defining a new abstraction. I was able to fix it by making `ActorTimer` and `ActorReminder` more beefy, and extracting base classes `ActorTimerToken`, `ActorReminderToken`.

All of the other types like `IActorReminder`, `TimerInfo`, `ReminderInfo` aren't that interesting for users. This is an example of how challenging it can be to carve out an abstraction after the fact 😉